crates/exex/exex/src/backfill.rs

```diff
@@ -0,0 +1,306 @@
+use reth_evm::execute::{BatchExecutor, BlockExecutionError, BlockExecutorProvider};
+use reth_exex_types::BackfillThresholds;
+use reth_node_api::FullNodeComponents;
+use reth_node_core::node_config::NodeConfig;
+use reth_primitives::BlockNumber;
+use reth_provider::{
+    BlockReader, Chain, DatabaseProviderFactory, HeaderProvider, HistoricalStateProviderRef,
+    ProviderError, StaticFileProviderFactory, TransactionVariant,
+};
+use reth_prune_types::PruneModes;
+use reth_revm::database::StateProviderDatabase;
+use reth_stages_api::format_gas_throughput;
+use reth_tracing::tracing::{debug, trace};
+use std::{
+    ops::RangeInclusive,
+    time::{Duration, Instant},
+};
+
+/// Factory for creating new backfill jobs.
+#[derive(Debug, Clone)]
+pub struct BackfillJobFactory<Node: FullNodeComponents> {
```

Comment:
this also don't needs the FullComponents trait, but we can keep a helper function that initializes from components.